src/modules/75-cf/pages/onboarding/__test__/SelectEnvironmentView.test.tsx

```diff
@@ -102,51 +85,129 @@ describe('SelectEnvironmentView', () => {
     })
   })
 
-  test('It should open the Create Environment modal', async () => {
+  test('It should open and close the Create Environment modal', async () => {
     renderComponent({
       language: {
         name: 'javascript',
         icon: 'javascripticon',
         type: PlatformEntryType.CLIENT,
         readmeStringId: 'cf.onboarding.readme.javascript'
-      },
-      selectedEnvironment: { identifier: 'foo', name: 'foo' },
-      apiKey: {
-        name: 'xxx-xxx-xxx',
-        apiKey: 'xxx-xxx-xxx',
-        identifier: 'xxx-xxx-xxx',
-        type: 'client'
       }
     })
 
     await waitFor(() => {
       expect(screen.getByText('cf.onboarding.selectOrCreateEnvironment')).toBeVisible()
       expect(screen.getByText('cf.onboarding.environmentDescription')).toBeVisible()
       expect(screen.queryByRole('button', { name: 'cf.environments.apiKeys.addKeyTitle' })).not.toBeInTheDocument()
-      expect(screen.getByText('cf.onboarding.keyDescriptionClient')).toBeVisible()
+      expect(screen.queryByText('cf.onboarding.keyDescriptionClient')).not.toBeInTheDocument()
+      expect(screen.getByRole('textbox', { name: 'cf.onboarding.selectOrCreateEnvironment' })).toBeInTheDocument()
+    })
+
+    const envInput = screen.getByRole('textbox', { name: 'cf.onboarding.selectOrCreateEnvironment' })
+
+    userEvent.click(envInput)
+
+    await waitFor(() => {
+      // one existing environment in dropdown
+      expect(document.getElementsByTagName('li')).toHaveLength(1)
+      expect(document.getElementsByTagName('li')[0]).toHaveTextContent('bar')
```

Comment:
*Not a blocker*. You can use `screen.getAllByRole('listitem')` instead of a query selector
```suggestion
      const items = screen.getAllByRole('listitem')
      expect(items).toHaveLength(1)
      expect(items[0]).toHaveTextContent('bar')
```